cranelift/codegen/src/machinst/vcode.rs

```diff
@@ -115,13 +121,16 @@ pub struct VCodeBuilder<I: VCodeInst> {
 
     /// Current basic block instructions, in reverse order (because blocks are
     /// built bottom-to-top).
-    bb_insns: SmallVec<[I; 32]>,
+    bb_insns: SmallVec<[(I, Option<SourceLoc>); 32]>,
 
     /// Current IR-inst instructions, in forward order.
-    ir_inst_insns: SmallVec<[I; 4]>,
+    ir_inst_insns: SmallVec<[(I, Option<SourceLoc>); 4]>,
 
     /// Start of succs for the current block in the concatenated succs list.
     succ_start: usize,
+
+    /// Current source location.
+    cur_srcloc: Option<SourceLoc>,
```

Comment:
There is already a SourceLoc reserved for when the source loc is unknown: SourceLoc::default().

Reply:
Ah, yes, this is much nicer (as well as the corresponding change to not-present indices on the regalloc side) -- done.